### PR TITLE
project lib restructure

### PR DESCRIPTION
**Pull request by gtalha07**
- Keep providers folder common for features with state notifiers residing in it to make better async pattern overall.
- correct some code files directory and typos.

### CODE REVIEW

**Review by bitfriend — APPROVED** [review]
Looks good to me

File: app/lib/common/providers/common_providers.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import 'package:acter/common/models/profile_data.dart';
```
Comment (gnunicorn) [inline app/lib/common/providers/common_providers.dart:1]: isn't this filename a bit too much of a catch-all? like account, network state and loading, chat and profile data are clearly not the same category of things ...

File: app/lib/router/router.dart [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import 'package:acter/common/dialogs/dialog_page.dart';
```
Comment (gnunicorn) [inline app/lib/router/router.dart:1]: I think this file-rename is also confusing. if anything it should be `router/routes.dart`, no?

**Review by gnunicorn — APPROVED** [review]
Overall fine with me, but two things I think we should change still. Can be done either in this PR or afterwards, though.

File: app/lib/common/providers/common_providers.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/common/providers/common_providers.dart:1]: Yes, I agree with not belonging to same categories but I named it as in the specifics of containing the providers which are global in application layout and is intended to be used in more than one features.

File: app/lib/router/router.dart (hunk 2 above)
Comment (gtalha07) [reply app/lib/router/router.dart:1]: Hmm, I have created another file with similar name which includes our route definitions in ```utils``` directory. This is more like our navigation function logic. How does that sounds?

**Pull request merged**